commit ba38a2c037c7900fb59bd36c51603cd7c2b0fb3f (refs/changes/10/4610/4)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2018-07-31T15:05:08-07:00 (1 year, 2 months ago)
    
    TRITON-647 vmapi NAT zone destruction should use SAPI DeleteInstance
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/lib/workflows/add-nics.js b/lib/workflows/add-nics.js
index cabb5c1..262f82a 100644
--- a/lib/workflows/add-nics.js
+++ b/lib/workflows/add-nics.js
@@ -23,7 +23,7 @@ var async;  // stub to keep jsl happy
 var common = require('./job-common');
 var fabricCommon = require('./fabric-common');
 
-var VERSION = '7.2.1';
+var VERSION = '7.2.2';
 
 
 /*
diff --git a/lib/workflows/destroy.js b/lib/workflows/destroy.js
index 4c39140..5d356b9 100644
--- a/lib/workflows/destroy.js
+++ b/lib/workflows/destroy.js
@@ -20,7 +20,7 @@ var common = require('./job-common');
 var fabricCommon = require('./fabric-common');
 var nfsVolumes = require('./nfs-volumes');
 
-var VERSION = '7.1.4';
+var VERSION = '7.1.5';
 
 
 /*
diff --git a/lib/workflows/fabric-common.js b/lib/workflows/fabric-common.js
index 5d03c78..53d95e6 100644
--- a/lib/workflows/fabric-common.js
+++ b/lib/workflows/fabric-common.js
@@ -487,10 +487,6 @@ function destroyFabricNats(job, cb) {
         url: napiUrl,
         headers: headers
     });
-    var vmapi = new sdcClients.VMAPI({
-        url: vmapiUrl,
-        headers: headers
-    });
 
     /* Find the NAT VM and destroy it. */
     function destroyNatZone(fabricNic, done) {
@@ -498,7 +494,14 @@ function destroyFabricNats(job, cb) {
             alias: 'nat-' + fabricNic.network_uuid,
             state: 'running'
         };
+        var vmapi = new sdcClients.VMAPI({
+            url: vmapiUrl,
+            headers: headers
+        });
+
         vmapi.listVms(listVmParams, function onGetVm(err, vms) {
+            var sapi;
+
             if (err) {
                 done(err);
                 return;
@@ -520,11 +523,13 @@ function destroyFabricNats(job, cb) {
                 'Destroying NAT zone');
 
             /* Synchronously destroy the NAT vm. */
-            var deleteVmParams = {
-                sync: true,
-                uuid: vms[0].uuid
-            };
-            vmapi.deleteVm(deleteVmParams, function (dErr) {
+            sapi = new sdcClients.SAPI({
+                url: sapiUrl,
+                headers: headers,
+                version: '~2',
+                log: job.log.child({ component: 'sapi' })
+            });
+            sapi.deleteInstance(vms[0].uuid, function (dErr) {
                 if (dErr && dErr.statusCode !== 404) {
                     done(dErr);
                     return;
diff --git a/lib/workflows/provision.js b/lib/workflows/provision.js
index c7a8267..17ccdb2 100644
--- a/lib/workflows/provision.js
+++ b/lib/workflows/provision.js
@@ -23,7 +23,7 @@ var nfsVolumes = require('./nfs-volumes');
 
 var wfapiUrl;
 
-var VERSION = '8.0.1';
+var VERSION = '8.0.2';
 
 
 /*
diff --git a/package.json b/package.json
index dc7a977..54d449d 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "vmapi",
   "description": "VMs API",
-  "version": "9.6.1",
+  "version": "9.6.2",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
